### PR TITLE
Implement Dynamic CORS Configuration (Fixes #2)

### LINKED ISSUE #2: Support for Configuring CORS Allowed Origins from .env File
### Summary

It would be useful to add support for configuring the CORS allowed origins dynamically from the `.env` file. This is particularly useful when running the backend in different environments (development, staging, production) with multiple origins.

### Current Behavior

Currently, allowed origins need to be manually set in the code using `CorsLayer`. This works, but it’s less flexible when deploying across multiple environments with different origins.

### Expected Behavior

The backend should be able to read a list of allowed origins from the `.env` file. The list would be defined as a comma-separated string in the `.env` file, which would then be parsed in the code and passed to `CorsLayer`.

### Proposed Solution

1. **Define allowed origins in the `.env` file** as a comma-separated list of origins,e.g.:
                `ALLOWED_ORIGINS="http://example.com,http://another-example.com,http://localhost:3000"`

2. **Load and parse the origins** in the backend code, split the string by commas, and create a vector of allowed origins.

3. **Configure the `CorsLayer`** using the parsed origins,

### PR DESCRIPTION
**Pull request by HarshitShukla-dev**
This pull request addresses issue #2 by introducing dynamic CORS configuration to the backend. The changes allow the application to read allowed origins from an environment variable defined in the `.env` file, enhancing flexibility for different deployment environments (development, staging, production).

#### **Key Changes:**
- **Dynamic Configuration**: Allowed CORS origins are now read from the `.env` file as a comma-separated string.
- **Parsing and Validation**: The origins are parsed and validated as `Uri` types, with any invalid URLs logged as warnings.
- **Type Handling**: Implemented conversion of `Uri` to `HeaderValue` for compatibility with the `AllowOrigin::exact()` method in the CORS configuration.
- **Error Handling**: Added error handling for potential parsing issues, ensuring that the application does not crash on invalid inputs.

### **Reminder:**
Please ensure to update the `.env` file with the `ALLOWED_ORIGINS` variable in the following format:

```
ALLOWED_ORIGINS="http://example.com,http://another-example.com,http://localhost:3000"
```

### CODE REVIEW

**Review by Sidharth-Singh10 — CHANGES_REQUESTED** [review]
@HarshitShukla-dev , load the environment variable in `constants.rs`. and then use the constant in `main.rs`.

**Comment by HarshitShukla-dev** [comment]
> @HarshitShukla-dev , load the environment variable in `constants.rs`. and then use the constant in `main.rs`.

I’m sorry for the oversight. I've successfully loaded the environment variable in constants.rs and integrated the constant into main.rs. The ALLOWED_ORIGINS variable is now dynamically configured based on the values specified in the .env file. Additionally, I've added error handling to ensure that the application gracefully handles any issues related to the environment variable.

If you have any further suggestions or need additional changes, please let me know!

Thanks!

File: .env.example [hunk 1]
```diff
@@ -5,4 +5,4 @@ SMTP_HOST="smtp.example.com"
 OTP_SECRET="lorememsum"
 JWT_SECRET="yoursecert"
 PASS_RESET_LINK="http://localhost:5173/reset-password"
-ALOOWED_ORIGIN="http://localhost:3000,http://yourdomain.com"
\ No newline at end of file
+ALLOWED_ORIGIN="http://localhost:3000,http://yourdomain.com"
```
Comment (Sidharth-Singh10) [inline .env.example]: The spelling here is different from what you used in `constant.rs`. 
Change it to `ALLOWED_ORIGINS`

File: src/utils/constants.rs [hunk 2]
```diff
@@ -6,23 +6,53 @@ lazy_static! {
...
+    pub static ref JWT_SECRET: String = {
+        env::var("JWT_SECRET").expect("JWT_SECRET must be set")
+    };
+
+    pub static ref PASS_RESET_LINK: String = {
+        env::var("PASS_RESET_LINK").expect("PASS_RESET_LINK must be set")
+    };
+
+    pub static ref ALLOWED_ORIGINS: Vec<String> = {
+        dotenv().ok();
```
Comment (Sidharth-Singh10) [inline src/utils/constants.rs]: calling `dotenv().ok()` again isn't necessary.

**Comment by Sidharth-Singh10** [comment]
If `ALLOWED_ORIGINS` is not set, do not default it to a empty list, exit the program gracefully with a warning.
Pointed out some minor faults please correct those.

**Comment by HarshitShukla-dev** [comment]
@Sidharth-Singh10, Fixed the issues.

**Comment by Sidharth-Singh10** [comment]
@HarshitShukla-dev , thanks for the work!!!

File: Cargo.toml [hunk 3]
```diff
@@ -20,7 +20,7 @@ chrono = "0.4.38"
 sea-orm = { version = "1.0.0-rc.5", features = [ "sqlx-postgres", "runtime-tokio-rustls", "macros" ] }
 uuid = { version = "1.10.0", features = ["v4"] }
 bcrypt = "0.15.1"
-tower-http = { version = "0.5.2", features = ["cors"] }
+tower-http = { version = "0.6.1", features = ["cors"] }
```
Comment (Sidharth-Singh10) [inline Cargo.toml:23]: please do not change versions of dependencies unless specified

**Comment by HarshitShukla-dev** [comment]
Hey @Sidharth-Singh10, it'll great if you assign the hacktoberfest and gssoc labels to both the pr and the issue.

**Pull request merged**